ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -246,6 +250,69 @@ public boolean updateAccounts(Collection<Account> accounts) {
     return updateAccountsWithAccountMetadataStore(accounts, accountMetadataStore);
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Pair<String, String> addContainer(String accountName, Container newContainer) {
+    checkOpen();
+    if (accountName == null || accountName.isEmpty() || newContainer == null || newContainer.getName() == null
+        || newContainer.getName().isEmpty()) {
+      logger.error("Account name: {} or container: {} is invalid", accountName, newContainer);
+      throw new IllegalArgumentException("Account or container name is null or empty");
+    }
+    Account account = getAccountByName(accountName);
+    if (account == null) {
+      logger.error("Account {} is not found", accountName);
+      throw new IllegalArgumentException("Account " + accountName + " is not found");
+    }
+    String accountId = String.valueOf(account.getId());
+    for (Container container : account.getAllContainers()) {
+      // make sure there is no conflict container (conflict means a container with same name but different attributes already exists).
+      if (container.getName().equals(newContainer.getName())) {
+        if (containerEditableFieldsAreSame(container, newContainer)) {
+          // If an exactly same container already exists, we directly return its id. Adding same container multiple times
+          // should be no-op.
+          return new Pair<>(accountId, String.valueOf(container.getId()));
```

Comment:
Why are the numerical IDs returned as a pair of strings instead of `Pair<Short, Short>`?

Reply:
I find containerId/accountId are `short` type, however, they are `int` in closed source code. Considering we may change the type to int in the future, so I want to use `String` here no matter what type of accountId/containerId are.

Reply:
Yes, we will need to change containerId to int.  AccountId can stay short; we may want to also make it int for consistency, if we don't run out of bits in blobId.

Reply:
Since you added it to config, can you make the method generic now?

Reply:
why not change this to `Pair<Integer, Integer>` then? That would fit both shorts and ints and allow for future flexibility.

I would like to point out that changing container ID to an int isn't the simplest change. There are implications in the blob ID, blob properties, and index value binary formats. A new version of each one will likely be required along with a staged rollout.

Reply:
I see the complexity of changing container id to `int`. For now, sure, I can change the return type to `Integer`.